plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/guru/VmwareVmImplementer.java

```diff
@@ -227,7 +227,7 @@ private void configureDomainRouterNicsAndDetails(VirtualMachineProfile vm, Virtu
             long networkId = publicNicProfile.getNetworkId();
             NetworkVO network = networkDao.findById(networkId);
 
-            for (; i < nics.length + vmwareMgr.getRouterExtraPublicNics(); i++) {
+            for (; i < nics.length; i++) {
```

Comment:
@harikrishna-patnala extra pubic nics configuration not required for domain router any more?

Reply:
yes @sureshanaparti, in the case of VMware the extra nic configuration is not there from now. I still see some references for HyperV, so I did not remove the configuration parameter completely.

Reply:
after upgrade, does this remove any existing extra nics in the domain router after restart?

Reply:
Yes, after restart any unused reserved nics will be removed. Any nics which are in use will continue to serve.